clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -963,9 +984,24 @@ private class HeartbeatReporterThread extends Thread {
     @Override
     public void run() {
       while (!Thread.interrupted()) {
-        for (Long lastHeartbeat: currentVersionLastHeartbeat.values()) {
-          changeCaptureStats.recordLag(System.currentTimeMillis() - lastHeartbeat);
+        int maxVersion = -1;
+        int minVersion = Integer.MAX_VALUE;
+        for (Map.Entry<Integer, Long> lastHeartbeat: currentVersionLastHeartbeat.entrySet()) {
```

Comment:
The key of `currentVersionLastHeartbeat` is partition number, not version number.

Reply:
fixed.